src/ZODB/Connection.py

```diff
@@ -757,7 +758,11 @@ def afterCompletion(self, transaction):
         #    finalizing previous ones without calling begin.  We pass
         #    False to avoid possiblyt expensive sync calls to not
         #    penalize well-behaved applications that call begin.
-        self.newTransaction(transaction, False)
+        if hasattr(self._storage, 'afterCompletion'):
+            self._storage.afterCompletion()
```

Comment:
> Are you referring to the lines above? or below? (I'm guessing below.)

Yeah. But I couldn't move the comment after I started it (this started out to be a different comment. :)

> We're only deferring in the explicit case.

True. Typo artifact in my comment.

> I would think that the cost of polling remote servers outweighs the benefits of maybe invalidating things.

Could be. I don't have good intuition about that.


> I agree that doing a cache gc here makes sense.

> I also wonder about the wisdom of doing cache gc at the beginning of a transaction. Perhaps the cache gc should be moved from new to here.

+1. Those both seem to make sense to me.